### PR TITLE
Improvement/zk UI 311 azure archive location

### PR DESCRIPTION
**Pull request by MonPote**

### CODE REVIEW

File: src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+          label="Temperature"
+          helpErrorPosition="bottom"
+          labelHelpTooltip={
+            <>
+              The Temperature of this Location is Cold. <br /> <br /> You can
+              move your data in this Location through a Transition Workflow.
+              <br /> <br />
+              Once your data are in this Location, you can only trigger a
+              request for restoration to get a temporary access to the object.
+            </>
```
Comment (JBWatenbergScality) [inline src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx]: nit: I feel like we have a component containing this tooltip info, would be great if we can reuse it

File: src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+              value={formState.bucketName}
+              autoComplete="off"
+              onChange={onFormItemChange}
+            />
+          }
+        />
+      </FormSection>
+      <FormSection title={{ name: 'Queue' }} forceLabelWidth={210}>
+        <FormGroup
+          id="tutu"
```
Comment (ChengYanJin) [inline src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx:260]: 😄

Comment (ChengYanJin) [inline src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx]: 😄

File: src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx [hunk 3]
```diff
@@ -0,0 +1,612 @@
...
+              type="text"
+              placeholder="Container Name"
+              value={formState.bucketName}
+              autoComplete="off"
+              onChange={onFormItemChange}
+            />
+          }
+        />
+      </FormSection>
+      <FormSection title={{ name: 'Queue' }} forceLabelWidth={210}>
```
Comment (ChengYanJin) [inline src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx:238]: To check why there is no separation line between the form sections.

File: src/react/backend/location/LocationDetails/__tests__/LocationDetailsAzureArchive.test.tsx [hunk 4]
```diff
@@ -0,0 +1,428 @@
+import { fireEvent, getByLabelText, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { debug } from 'jest-preview';
```
Comment (ChengYanJin) [inline src/react/backend/location/LocationDetails/__tests__/LocationDetailsAzureArchive.test.tsx]: can be removed.

File: src/react/backend/location/LocationDetails/__tests__/LocationDetailsAzureArchive.test.tsx [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+
+    textGrep.forEach((text) => {
+      expect(screen.getByLabelText(text)).toBeInTheDocument();
+      expect(screen.getByLabelText(text)).toHaveValue('');
+    });
+  });
+
+  // usefull for later
+
+  test('Azure Service Bus Topic + Client Secret', async () => {
```
Comment (ChengYanJin) [inline src/react/backend/location/LocationDetails/__tests__/LocationDetailsAzureArchive.test.tsx:231]: We should also check the `Azure Shared Key` is disable for the `Azure Service Bus Topic`.

File: src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx [hunk 6]
```diff
@@ -0,0 +1,612 @@
...
+              The Temperature of this Location is Cold. <br /> <br /> You can
+              move your data in this Location through a Transition Workflow.
+              <br /> <br />
+              Once your data are in this Location, you can only trigger a
+              request for restoration to get a temporary access to the object.
+            </>
+          }
+          content={
+            <Stack>
+              <ColdStorageIcon /> <TertiaryText>Cold</TertiaryText>
```
Comment (ChengYanJin) [inline src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx]:
```suggestion
              <ColdStorageIcon /> <Text color="textSecondary">Cold</Text>
```

File: src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx [hunk 7]
```diff
@@ -0,0 +1,612 @@
...
+          id="tutu"
+          label="Queue type"
+          helpErrorPosition="bottom"
+          required
+          content={
+            <Select
+              placeholder="Select an option..."
+              onChange={onChangeQueueType}
+              value={formState.queue.type}
+              inputId="tutu"
```
Comment (JBWatenbergScality) [inline src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx]: We probably want to use id here

File: src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx [hunk 8]
```diff
@@ -0,0 +1,612 @@
...
+              label="Queue Endpoint"
+              id="queue.endpoint"
+              helpErrorPosition="bottom"
+              required
+              content={
+                <Input
+                  name="queue.endpoint"
+                  id="queue.endpoint"
+                  type="text"
+                  placeholder="sb://toto"
```
Comment (JBWatenbergScality) [inline src/react/backend/location/LocationDetails/LocationDetailsAzureArchive.tsx]: Oops, I would not use "toto" as a public example

**Review by JBWatenbergScality — APPROVED** [review]

**Comment by MonPote** [comment]
ping

**Pull request merged**